parlai/mturk/tasks/acute_eval/run.py

```diff
@@ -0,0 +1,544 @@
+    response_data = save_data['worker_data'][worker_id]['response']['task_data']
+    num_onboarding_tasks = 0
+    num_correct = 0
+    for i in range(len(task_data)):
+        task_specs = task_data[i]['task_specs']
+        if not task_specs.get('is_onboarding', False):
+            continue
+        worker_response = float(response_data[i]['speakerChoice'])
+        expected_response = (
+            1
+            if (
+                (
+                    task_specs['conversation_order'] == [1, 0]
+                    and not task_specs['correctness_is_flipped']
+                )
+                or (
+                    task_specs['conversation_order'] == [0, 1]
+                    and task_specs['correctness_is_flipped']
+                )
+            )
+            else 2
+        )
+        num_onboarding_tasks += 1
+        if worker_response == expected_response:
+            num_correct += 1
+    if num_onboarding_tasks == 0:
+        if worker_id in onboarding_failed_workers:
+            return_task_data(worker_id, task_data)
+        return
+    if (num_correct / num_onboarding_tasks) >= threshold:
+        return
+    mturk_manager.soft_block_worker(worker_id)
+    onboarding_failed_workers.append(worker_id)
+
+
+def main(opt):
+    """Handles setting up and running a ParlAI-MTurk task by instantiating
+    an MTurk manager and configuring it for the qa_data_collection task
+    """
+    np.random.seed(opt['seed'])
+
+    # Set the task name to be the folder name
+    opt['task'] = os.path.basename(os.path.dirname(os.path.abspath(__file__)))
+
+    # append the contents of task_config.py to the configuration
+    opt.update(task_config)
+
+    display_agent_name = 'RatingWorker'
+
+    # Instantiate an MTurkManager with the given options and a maximum number
+    # of agents per world of 1 (based on the length of mturk_agent_ids)
+    mturk_manager = StaticMTurkManager(opt=opt)
+
+    mturk_manager.setup_server(
+        task_directory_path=os.path.dirname(os.path.abspath(__file__))
+    )
+
+    mturk_manager.set_onboard_function(onboard_function=None)
+
+    if opt['block_on_onboarding_fail'] and opt['block_qualification'] is None:
+        raise Exception(
+            "Set block_qualification or set block_on_onboarding_fail to False"
+        )
+
+    setup_task_queue(opt)
+
+    try:
+        # Initialize run information
+        mturk_manager.start_new_run()
+        # Set up the sockets and threads to recieve workers
+        mturk_manager.ready_to_accept_workers()
+
+        # Create the hits as specified by command line arguments
+        mturk_manager.create_hits()
+
+        def check_worker_eligibility(worker):
+            return True
+
+        def assign_worker_roles(workers):
+            workers[0].id = display_agent_name
+
+        def run_conversation(mturk_manager, opt, workers):
+            task_data = get_new_task_data(workers[0], opt['comparisons_per_hit'])
```

Comment:
`comparisons_per_hit` is not an argument above, so this causes an error. did you mean `subtasks` per hit?